### DISCUSSION TITLE
Add a SQLModel based table with CRUD feat

### DISCUSSION THREAD

**Discussion by KimigaiiWuyi** [zauberzeug/nicegui#383] [Ideas / Feature Requests]
quickly generate a visual dashboard about database admin WebUI（Includes the most basic CRUD feat）
May be inspired by [fastapi_amis_admin](https://github.com/amisadmin/fastapi_amis_admin)  （Fast page generation based on sqlmodel model）
I know it can be complicated, but this feat would be very powerful！

**Comment by falkoschindler** [comment]
This sounds like an _application_ for NiceGUI, not a _feature_.
Although it would use and build upon NiceGUI, it would better be its own project in its own repository, don't you think?

**Reply by KimigaiiWuyi** [reply]
I agree with you, in fact I have started to try to develop it, but I have encountered a problem, the AG grid does not seem to have any width, for example if I put it under tabs, he will not see it at all.

I'm very sorry, I don't have any front-end knowledge, maybe need to set some css?like use `.classes()`

Anyway, here's my code

```python
from nicegui import ui

with ui.tabs() as tabs:
    ui.tab('Home', icon='home')
    ui.tab('About', icon='info')

with ui.tab_panels(tabs, value='Home'):
    with ui.tab_panel('Home'):
        ui.video('https://test-videos.co.uk/vids/bigbuckbunny/mp4/h264/360/Big_Buck_Bunny_360_10s_1MB.mp4')
    with ui.tab_panel('About'):
        table = ui.table({
            'columnDefs': [
                {'headerName': 'Name', 'field': 'name','editable':True,'resizable':True},
                {'headerName': 'Age', 'field': 'age','editable':True,'resizable':True},
            ],
            'rowData': [
                {'name': 'Alice', 'age': 18},
                {'name': 'Bob', 'age': 21},
                {'name': 'Carol', 'age': 42},
            ],
            'rowSelection': 'multiple',
        })
ui.run()
```

**Reply by falkoschindler** [reply]
The table has a default width of 100%, i.e. the width of the parent container. But the panel seems to collapse to a width of 0 because it has no child element with non-zero minimum width.
You can either set a width for the `ui.table`, the `ui.tab_panel` or the `ui.tab_panels`, e.g. with `.style('width: 500px')`.
